### PR TITLE
Airlock sabotage negative station trait

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

A very simple negative station trait, with a low weight, that makes about 25% of the station's airlocks have random cut wires. The wires that get cut are based on a probability list, and generally the worst ones happen less often.

## Why It's Good For The Game

A new annoyance for crewmembers to contend with, once in a blue moon.

## Changelog

:cl: Wholesome chungus chonker 100
add: Airlock sabotage station trait
/:cl:

### CODE REVIEW

File: code/datums/station_traits/negative_traits.dm [hunk 1]
```diff
@@ -85,6 +85,45 @@
...
+	. = ..()
+	RegisterSignal(SSatoms, COMSIG_SUBSYSTEM_POST_INITIALIZE, PROC_REF(sabotage_airlocks))
+
+/datum/station_trait/airlock_sabotage/proc/sabotage_airlocks()
+	SIGNAL_HANDLER
+	var/datum/wires/airlock/airlock_wires
+	for(var/obj/machinery/door/airlock/airlock as anything in GLOB.airlocks)
+		if(!(is_station_level(airlock.z) && prob(25)))
+			continue
+		airlock_wires = airlock.wires
```
Comment (Mothblocks) [inline code/datums/station_traits/negative_traits.dm:122]: I don't think this is necessary or helpful, just makes the code a bit disjointed

Comment (ChungusGamer666) [reply code/datums/station_traits/negative_traits.dm:122]: eh, true, i'll just make it use airlock.wires

**Comment by ChungusGamer666** [comment]
> If it can cut bolt wires then can't it lead to softlocking players at roundstart? Ex: jani spawns in custodial closet, door spawns bolted (i guess the jani would need to use disposals?)

Well, there is a chance, even if small.
In that case they'll have to rely on the kindness of a good samaritan who can open a nearby airlock for them. They do have a radio headset on their ears.

**Comment by Overuseddart** [comment]
Mfw when the the captain spawns in a bolted room and is unable to get out. The mime started in an electrified room, and the SM door released sparks and activated the engine.

**Comment by ChungusGamer666** [comment]
Everything seems to be working as intended, quite a few airlocks get affected but not enough to ruin the entire round.

**Comment by private-tristan** [comment]
> Mfw when the the captain spawns in a bolted room and is unable to get out. The mime started in an electrified room, and the SM door released sparks and activated the engine.

solutions for these:
> captain has command door remote, can unbolt door and leave
mime can throw their ID at the electrified door, causing it to be opened
supermatter wouldn't be activated by sparks unless there's somehow plasma in the SM room.

**Comment by Overuseddart** [comment]
> solutions for these:
> 
> > captain has command door remote, can unbolt door and leave

The other two are solutions but the captain doesnt actually spawn with his locker on some maps, so the captain could theoretically be trapped

**Comment by ChungusGamer666** [comment]
> 
> The other two are solutions but the captain doesnt actually spawn with his locker on some maps, so the captain could theoretically be trapped

still a non-issue unless somehow the station also starts with no tcomms

**Comment by Overuseddart** [comment]
> > The other two are solutions but the captain doesnt actually spawn with his locker on some maps, so the captain could theoretically be trapped
> 
> still a non-issue unless somehow the station also starts with no tcomms

that is true,
Does this affect other Z-Levels besides the station one? Ruins and the syndicate base have wires as well.

**Comment by ChungusGamer666** [comment]
> 
> that is true, Does this affect other Z-Levels besides the station one? Ruins and the syndicate base have wires as well.

Only station airlocks are affected, everything else is left as is.

**Comment by Overuseddart** [comment]
> > that is true, Does this affect other Z-Levels besides the station one? Ruins and the syndicate base have wires as well.
> 
> Only station airlocks are affected, everything else is left as is.

Cool, this trait sounds fun!

**Comment by Farquaar** [comment]
Imagine starting your first round of SS13 and wondering why you can't leave the janitors closet

**Comment by YakumoChen** [comment]
Needs to hard preclude Bolt Wire, and have a safety against cutting all power wires

I think what everyone but Bob is missing here is that these are _cut_ wires. Having your only door out of a room bolt-cut is something only an engineer is going to be able to fix roundstart, precluding the lucky spawn next to a toolbox with help from the AI.

Having someone softlocked into a room at roundstart is definitely not good for gameplay.

**Comment by ChungusGamer666** [comment]
> Needs to hard preclude Bolt Wire, and have a safety against cutting all power wires
> 
> I think what everyone but Bob is missing here is that these are _cut_ wires. Having your only door out of a room bolt-cut is something only an engineer is going to be able to fix roundstart, precluding the lucky spawn next to a toolbox with help from the AI.
> 
> Having someone softlocked into a room at roundstart is definitely not good for gameplay.

It's not softlocking. This is literally a multiplayer roleplaying game where crewmembers are encouraged to help each other survive on the station. Having to ask for help on the radio isn't "softlocking".
Additionally, the chance of a door being bolted is extremely low, only 25% of airlocks on-station get wires cut, and only 30% of the affected airlocks will be bolted. 
For every airlock on-station, the chance of it being bolted when this trait is applied, a whopping 7.5%. Considering most rooms have more than one airlock as a way in and out, being softlocked is at best a minor concern.

**Comment by YakumoChen** [comment]
> It's not softlocking. This is literally a multiplayer roleplaying game where crewmembers are encouraged to help each other survive on the station. Having to ask for help on the radio isn't "softlocking".

Starting the game being unable to play the game because of a door that needs a full toolbox + multitool to leave is not what I call good game design. Being able to use the radio does not guarantee your ability to leave in any reasonable amount of time either.

**Comment by ChungusGamer666** [comment]
> Starting the game being unable to play the game because of a door that needs a full toolbox + multitool to leave is not what I call good game design. Being able to use the radio does not guarantee your ability to leave in any reasonable amount of time either.

Shame, this is a game all about disasters unfolding around you and bouts of bad luck like being hit by a meteor and dying instantly (has happened to me once, maybe twice) can and will happen. Getting "softlocked" by the airlocks should be rare enough that I don't want to remove functionality from this PR just to prevent this one extremely rare misfortune.

Also this is straight up a lie, assuming only the bolt wire was cut, you only need 2 tools to escape, a screwdriver to open the panel and a wirecutter to mend the wire. You will get shocked once, but you'll be able to leave just fine after that.

**Comment by Overuseddart** [comment]
> Shame, this is a game all about disasters unfolding around you and bouts of bad luck like being hit by a meteor and dying instantly (has happened to me once, maybe twice) can and will happen. Getting "softlocked" by the airlocks should be rare enough that I don't want to remove functionality from this PR just to prevent this one extremely rare misfortune.
> 
> Also this is straight up a lie, assuming only the bolt wire was cut, you only need 2 tools to escape, a screwdriver to open the panel and a wirecutter to mend the wire. You will get shocked once, but you'll be able to leave just fine after that.

There’s maps in /TG/ rotation that has a spawn point in maints for assistants, one door out, no cameras, maybe make it so it only electrifies maints?

**Comment by YakumoChen** [comment]
> Also this is straight up a lie, assuming only the bolt wire was cut, you only need 2 tools to escape, a screwdriver to open the panel and a wirecutter to mend the wire. You will get shocked once, but you'll be able to leave just fine after that.

Cutting the bolt wire drops the bolts, so mending the wire is still going to leave you locked in a bolted room.

**Comment by YakumoChen** [comment]
That doesn't even account for if the door is shocked, which entirely prevents you from fixing it, much less opening it.

**Comment by Mothblocks** [comment]
I think it's fine if it bolts doors but we definitely shouldn't have any cases where you can't get out of your spawn location without someone else

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I guess I'll stop on that for now, but the code looks fine

**Comment by ChungusGamer666** [comment]
> I guess I'll stop on that for now, but the code looks fine

If I make the doors bolt open instead of bolt closed, is that more acceptable?

**Comment by YakumoChen** [comment]
Still doesn't change the possibility that doors could be shocked or depowered, leading to the same problem

**Comment by Mothblocks** [comment]
Mmm yeah that is true. You could whitelist doors but that would be a nightmare for mappers and would definitely regress. I'm not sure the best solution for this

**Comment by calliophis-maculiceps** [comment]
>  > solutions for these:
> 
> > 
> 
> > > captain has command door remote, can unbolt door and leave
> 
> 
> 
> The other two are solutions but the captain doesnt actually spawn with his locker on some maps, so the captain could theoretically be trapped
> 
> 
> 
> 

does the captain's spawnpoint not have access to disposals?

**Comment by ChungusGamer666** [comment]
> Still doesn't change the possibility that doors could be shocked or depowered, leading to the same problem

Shocked doors can still be opened if memory serves me correctly. So, not really.

**Comment by ChungusGamer666** [comment]
> Mmm yeah that is true. You could whitelist doors but that would be a nightmare for mappers and would definitely regress. I'm not sure the best solution for this

Due to what I said in my last comment, making doors get bolted open (unless they already have bolts down) is most likely going to be okay.
In the case of a depowered door, it can't shock you and only requires a crowbar to open (which means pretty much any bystander can help if you cry on the radio), on top of being extremely rare due to sheer probability.
I don't know.

**Comment by YakumoChen** [comment]
"Get out without someone else" was the deciding factor here.

To be honest I think the trait is just overall really bad for the game. Having the doors be invisibly dangerous or wonky at roundstart is too trollish and is especially harmful to people new at the game.

**Comment by ChungusGamer666** [comment]
> "Get out without someone else" was the deciding factor here.
> 

Space Station 13 is a multiplayer roleplaying game. I have nothing else to say. If you genuinely think the game should be designed with singleplayer gameplay in mind, that is your problem. I would address the rest of the comment but at this point I am repeating myself to someone who genuinely thinks crewmembers should be self sufficient.

**Comment by YakumoChen** [comment]
> I think it's fine if it bolts doors but we definitely shouldn't have any cases where you can't get out of your spawn location without someone else

?

**Comment by ChungusGamer666** [comment]
> > I think it's fine if it bolts doors but we definitely shouldn't have any cases where you can't get out of your spawn location without someone else
> 
> ?

my bad im rarted

**Comment by ChungusGamer666** [comment]
yeah whatever i dont really wanna compromise until this trait becomes a pointless non-issue, maybe i'll come back to this later after having a revelation in my dremas or somethinf but fuck this shit for now

**Comment by Mothblocks** [comment]
I have some insane ideas like trying to perform a JPS pathfind to make sure that any spawn always has an out but man would that be annoying to do :P

**Pull request closed without merging**